feature_engine/datetime/datetime_subtraction.py

```diff
@@ -0,0 +1,349 @@
+        )
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.reference_variables_)
+            _check_contains_na(X, self.variables_to_combine_)
+
+        # create all of the variable pairs
+        # exclude any pairs containing the same variable
+        variable_pairs = \
+            [(var1, var2) for var1, var2 in
+             itertools.product(self.reference_variables_, self.variables_to_combine_)
+             if var1 != var2]
+
+        if self.dedupe_variable_pairs:
+            # sort the tuple values and dedupe them
+            # use dict.fromkeys instead of set to make the order of the tuples
+            # deterministic
+            variable_pairs = \
+                cast(
+                    List[Tuple[Union[str, int], Union[str, int]]],
+                    list(dict.fromkeys(tuple(sorted([var1, var2])) for var1, var2
+                                       in variable_pairs))
+                )
+
+        if self.new_variables_names:
+            if len(self.new_variables_names) != len(variable_pairs):
+                raise ValueError(f'The number of new variable names '
+                                 f'({len(self.new_variables_names)}) must match the'
+                                 f'number of variable pairs ({len(variable_pairs)})')
+
+        self.variable_pairs_ = variable_pairs
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Subtract the groups of datetime variables.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns
+        -------
+        X_new: Pandas dataframe, shape = [n_samples, n_features + n_operations]
+            The dataframe with the new variables.
+        """
+
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check that input is a dataframe
+        X = _is_dataframe(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_input_matches_training_df(X, self.n_features_in_)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.reference_variables_)
+            _check_contains_na(X, self.variables_to_combine_)
+
+        # convert datetime variables
+        datetime_variables = set(self.reference_variables_ +
+                                 self.variables_to_combine_)
+
+        datetime_df = X[datetime_variables].apply(pd.to_datetime,
+                                                  dayfirst=self.dayfirst,
+                                                  yearfirst=self.yearfirst,
+                                                  utc=self.utc
+                                                  )
+
+        # find any non-datetime columns
```

Comment:
this was already done in fit. We do not need to repeat here.

in the transform we need this block of code: https://github.com/feature-engine/feature_engine/blob/main/feature_engine/creation/combine_with_reference_feature.py#L267 adapted for datetime variables.